### PR TITLE
Elasticsearch: Fix using multiple bucket script aggregations when only grouping by terms

### PR DESCRIPTION
**Pull request by MarceloNunesAlves**
Problem:

When you add more than one bucket script in the query it shows only one column.

problem

Solution:

In the parser when it was configurate more than one bucket it put the formula in name column. The same process in the groupby date histogram.

solution

### CODE REVIEW

**Comment by MarceloNunesAlves** [comment]
I've signed

**Review by lefranchi — APPROVED** [review]

**Comment by MarceloNunesAlves** [comment]
I need it to be reviewed to merge. This correction is very important to someone use elasticsearch.

**Comment by aocenas** [comment]
@MarceloNunesAlves thanks for the contribution and sorry for the late review. I am checking the issue with current master and I think this was already solved by updates to the data logic in Grafana. Right now what I get:

Screenshot from 2020-05-22 15-09-21

with this transform to join all the dataframes:
Screenshot from 2020-05-22 15-10-12

Let me know if this also solves your usecase.

**Comment by MarceloNunesAlves** [comment]
@aocenas Ok! But change Date Histogram to terms.

**Comment by aocenas** [comment]
Ah right did not notice that. It would be great though if those 2 modes gave you the same column names. Seems like this part of code is responsible for interpolating the script variable names https://github.com/grafana/grafana/blob/8d81aab13d3a2ce0b292c03cacdfdf349cfe2be1/public/app/plugins/datasource/elasticsearch/elastic_response.ts#L268-L277. All the info needed there seems to be available in the `processAggregationDocs` method so it seems feasible to extract that into separate function and reuse it there to get consistent naming.

Do you have time to try something like that?

**Comment by MarceloNunesAlves** [comment]
@aocenas, 
I tried to use the field "Select field" to alias but it didn't work. In this moment I don't have time but it solution solve the problem.

**Comment by MarceloNunesAlves** [comment]
Can't approve the fix?

**Comment by MarceloNunesAlves** [comment]
@torkelo, could you review my commit?

**Comment by torkelo** [comment]
I think @aocenas comment is correct, the same naming logic should be used for both date histogram (time series) result and table results (no date histogram)

**Comment by aocenas** [comment]
@torkelo Even though fixing the naming would be ideal I think we can merge this (after adding some unit test). It fixes an issue so it adds a value as it is and reusing the code that does the naming needs some refactoring so we could do that separately.

**Comment by torkelo** [comment]
ok, agree. Then it's only a test that would be good, and make sure this code is matched on the backend implementation

**Comment by MarceloNunesAlves** [comment]
Hey guys, done!

**Review by davkal — DISMISSED** [review]
Thanks for adding the test.

Do you mind patching https://github.com/grafana/grafana/blob/master/pkg/tsdb/elasticsearch/response_parser.go#L376 ?

**Comment by MarceloNunesAlves** [comment]
Ok! Thanks @davkal

**Comment by MarceloNunesAlves** [comment]
@davkal, done! tks

File: pkg/tsdb/elasticsearch/response_parser.go
Comment (marefr) [inline pkg/tsdb/elasticsearch/response_parser.go:379]:
```suggestion
						metricName = metric.Settings.MustString("script")
```

**Review by marefr — CHANGES_REQUESTED** [review]
See suggestion

**Comment by MarceloNunesAlves** [comment]
@marefr, I've removed because this code didn't return what it needed.

**Comment by MarceloNunesAlves** [comment]
@marefr, this problem:

grafana_bug

File: pkg/tsdb/elasticsearch/response_parser.go
Comment (marefr) [reply pkg/tsdb/elasticsearch/response_parser.go:379]: Sorry, this was what it should have been.

```suggestion
						metricName = metric.Settings.Get("script").MustString("")
```

**Comment by MarceloNunesAlves** [comment]
@marefr, done! tks

**Comment by MarceloNunesAlves** [comment]
@marefr/@davkal/@torkelo, is it ok now?

**Review by marefr — APPROVED** [review]
LGTM

**Comment by MarceloNunesAlves** [comment]
@marefr, thank you so much!

**Comment by aocenas** [comment]
@MarceloNunesAlves thanks again for the contribution :+1:

**Pull request merged**
